homeassistant/components/file/notify.py

```diff
@@ -29,14 +33,17 @@
 )
 
 
-def get_service(
+async def async_get_service(
     hass: HomeAssistant,
     config: ConfigType,
     discovery_info: DiscoveryInfoType | None = None,
-) -> FileNotificationService:
+) -> FileNotificationService | None:
     """Get the file notification service."""
-    filename: str = config[CONF_FILENAME]
-    timestamp: bool = config[CONF_TIMESTAMP]
+    if discovery_info is None:
+        # We only set up through discovery
+        return None
+    filename: str = discovery_info[CONF_FILENAME]
+    timestamp: bool = discovery_info.get(CONF_TIMESTAMP, False)
```

Comment:
```suggestion
    timestamp: bool = discovery_info[CONF_TIMESTAMP]
```
Has default value so will always be set

Reply:
Has a default value also in the config entry schema so it will always be set https://github.com/home-assistant/core/pull/116861/files#diff-d87a73cf82f310d78d33db7f066dae382f615cd34899289b61682666128a7448R47